app/helpers/reactjs_helper.rb

```diff
@@ -5,6 +5,25 @@ def mount_react_component(name, selector, data = [], opts = {})
     end
   end
 
+  # Mount react component in views
+  # Params:
+  # +name+:: the component name from the componentRegistry
+  # +data+:: data to pass to the component as props
+  #          available values: Hash, json-string, nil
+  # +opts[:flatten_data]+:: false = will pass the props to the component props.data
+  #                         true = will pass the props directly to the component props
+  def react_component(name, data = [], opts = {})
```

Comment:
it would be better to use kwarg here for `flatten_data` instead of a generic opts object. tbh, i'd even prefer getting rid of that option at least here and deciding on an opinionated way of passing data to components - if the component want some data wrapped in a `data` prop, the caller should wrap the data in that way, which it would need to know about anyways to decide if flatten_data should be true or false. 
Even if we still need it on the react side, we don't have to support it here as well - iirc this was just some workaround because we were inconsistent in the past regarding passing both data and props to components in different ways.

Reply:
maybe even rename the parameter to props so it's clear this will be the component's props.

Reply:
I've done this one and I believe the rest is related. Is it better like this? :)